### PR TITLE
Improve fork compatibility

### PR DESCRIPTION
**Pull request by BobDotCom**
This allows base command to discover forks as well. Forks that change the import name will obviously not work, but that's their fault.
No breaking changes here, just improvements.

## Rationale

This just lets the base command find the name of the discord module instead of setting it to discord.py. 

Before:

Screen Shot 2021-11-16 at 4 53 29 PM

After:

Screen Shot 2021-11-16 at 4 54 30 PM

## Summary of changes made

1. Adds `importlib_metadata>=3.7.0` to the requirements. The required function, `packages_distributions`, was [added in 3.7](https://github.com/python/importlib_metadata/blob/main/CHANGES.rst#v370).
2. Changed the package name/version in the root command to be discovered from the package name instead of statically "discord.py"

## Checklist

- [x] This PR changes the jishaku module/cog codebase
    - [ ] These changes add new functionality to the module/cog
    - [x] These changes fix an issue or bug in the module/cog
    - [x] I have tested that these changes work on a production bot codebase
    - [x] I have tested these changes against the CI/CD test suite
    - [ ] ~~I have updated the documentation to reflect these changes~~ No need, only bugfixes
- [ ] This PR changes the CI/CD test suite
    - [ ] I have tested my suite changes are well-formed (all tests can be discovered)
    - [ ] These changes adjust existing test cases
    - [ ] These changes add new test cases
- [ ] This PR changes prose (such as the documentation, README or other Markdown/RST documents)
    - [ ] I have proofread my changes for grammar and spelling issues
    - [ ] I have tested that any changes regarding Markdown/RST syntax result in a well formed document

### CODE REVIEW

File: jishaku/features/root_command.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 import discord
 from discord.ext import commands
+from importlib_metadata import packages_distributions
```
Comment (Gorialis) [inline jishaku/features/root_command.py:20]:
```suggestion
try:
    from importlib.metadata import packages_distributions
except ImportError:
    from importlib_metadata import packages_distributions
```

File: requirements/_.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 braceexpand>=0.1.7
 click>=8.0.1
 import_expression>=1.0.0,<2.0.0
+importlib_metadata>=3.7.0
```
Comment (Gorialis) [inline requirements/_.txt:4]:
```suggestion
importlib_metadata>=3.7.0; python_version<"3.10"
```

**Review by Gorialis — CHANGES_REQUESTED** [review]

**Comment by BobDotCom** [comment]
Changes implemented

**Comment by Gobot1234** [comment]
This version constrain is wrong it should be for <3.8 not 3.10

**Comment by BobDotCom** [comment]
> This version constrain is wrong it should be for <3.8 not 3.10

`importlib.metadata` was added to the python builtins in 3.8 with features from importlib_metadata version 1.4. Since we need a feature from importlib_metadata 3.7, python 3.10 is correct because importlib_metadata 4.4 features were added to python in 3.10.
Screen Shot 2021-11-18 at 9 21 09 AM

**Comment by Gobot1234** [comment]
Sorry I completely misread the version constraint not realising that the version numbers weren't related to python versions

**Pull request merged**